File: lib/models/subscription_plan.dart
class SubscriptionPlan {
  final String id;
  final String name;
  final double price;
  final String duration;
  final List<String> features;
  final bool isActive;

  SubscriptionPlan({
    required this.id,
    required this.name,
    required this.price,
    required this.duration,
    required this.features,
    required this.isActive,
  });

  factory SubscriptionPlan.fromJson(Map<String, dynamic> json) {
    return SubscriptionPlan(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      duration: json['duration'] ?? '',
      features: List<String>.from(json['features'] ?? []),
      isActive: json['isActive'] ?? true,
    );
  }
}

File: backend/models/SubscriptionPlan.js
const mongoose = require('mongoose');

const subscriptionPlanSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true
  },
  price: {
    type: mongoose.Schema.Types.Mixed,
    required: true,
    get: function(value) {
      return typeof value === 'string' ? parseFloat(value) : value;
    }
  },
  duration: {
    type: String,
    required: true,
    enum: ['monthly', 'yearly']
  },
  features: [{
    type: String,
    required: true
  }],
  isActive: {
    type: Boolean,
    default: true
  }
}, {
  timestamps: true,
  toJSON: { getters: true },
  toObject: { getters: true }
});

module.exports = mongoose.model('SubscriptionPlan', subscriptionPlanSchema, 'subscriptions');